File: cs/Day1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.IO;

namespace cs
{
    class Day1
    {
        public static void Part1()
        {
            int last = -1, count = 0;

            File.ReadAllLines(@"../../../../../data/day1.txt").Select(line => int.Parse(line)).ToList().ForEach(num =>
            {
                if (last > 0 && num > last)
                    count++;

                last = num;
            });

            Console.WriteLine("Day:  1 | Part: 1 | Result: " + count);
        }

        public static void Part2()
        {
            List<string> lines = File.ReadAllLines(@"../../../../../data/day1.txt").ToList();

            int last = -1, count = 0;

            for (int index = 0; index < lines.Count - 2; index++)
            {
                int sum = int.Parse(lines[index]) + int.Parse(lines[index + 1]) + int.Parse(lines[index + 2]);

                for (int off = 0; off < 3; off++) sum += int.Parse(lines[index + off]);

                if (last > 0 && sum > last)
                    count++;

                last = sum;
            }

            Console.WriteLine("Day:  1 | Part: 2 | Result: " + count);
        }
    }
}

File: php/Day1.php
<?php
    require_once('Utils.php');

    function Day1_Part1() {
        $last = -1;
        $count = 0;

        foreach (read_file_lines("../data/day1.txt", "r") as $line) {
            if ($last > 0 && (int) $line > $last)
                $count++;

            $last = (int) $line;
        }

        echo "Day:  1 | Part: 1 | Result: " . $count . "<br>";
    } 

    
    function Day1_Part2() {
        $file = fopen("../data/day1.txt", "r");
        $lines = read_file_lines("../data/day1.txt", "r");
        
        $last = -1;
        $count = 0;

        for ($index = 0; $index < count($lines) - 2; $index++) {
            $sum = $lines[$index] + $lines[$index + 1] + $lines[$index + 2];

            if ($last > 0 && $sum > $last)
                $count++;

            $last = $sum;
        }

        echo "Day:  1 | Part: 2 | Result: " . $count . "<br>";
    } 

    Day1_Part1();
    Day1_Part2();
    